lib/jsdom/living/nodes/HTMLInputElement-impl.js

```diff
@@ -406,6 +541,122 @@ class HTMLInputElementImpl extends HTMLElementImpl {
     }
   }
 
+  // https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasdate
+  get valueAsDate() {
+    if (!allowValueAsDate(this.type)) {
+      return null;
+    }
+
+    return this._parsedValue;
+  }
+
+  set valueAsDate(v) {
+    if (!allowValueAsDate(this.type)) {
+      throw new DOMException("The object is in an invalid state.", "InvalidStateError");
+    }
+    if (v !== null && !(v instanceof Date)) {
+      throw new TypeError("Failed to set value as Date");
+    }
+
+    if (v === null || (v instanceof Date && isNaN(v))) {
+      this._value = "";
+    }
+
+    const convertDateToString = this._convertDateToString;
+    this._value = convertDateToString(v);
+  }
+
+  get valueAsNumber() {
+    // https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasnumber
+    if (!allowValueAsNumber(this.type)) {
+      return NaN;
+    }
+
+    return this._parsedValue;
+  }
+
+  set valueAsNumber(v) {
+    if (!isFinite(v)) {
+      throw new TypeError("Failed to set infinite value as Number");
+    }
+
+    if (!allowValueAsNumber(this.type)) {
+      throw new DOMException("The object is in an invalid state.", "InvalidStateError");
+    }
+
+    const convertNumberToString = this._convertNumberToString;
+    this._value = convertNumberToString(v);
+  }
+
+  _stepUpdate(n, isUp) {
+    // https://www.w3.org/TR/html52/sec-forms.html#dom-htmlinputelement-stepup
+
+    if (!allowValueAsNumber(this.type)) {
+      throw new DOMException("The object is in an invalid state.", "InvalidStateError");
+    }
+
+    const allowedValueStep = this._allowedValueStep;
+    if (allowedValueStep === null) {
+      throw new DOMException("The object is in an invalid state.", "InvalidStateError");
+    }
+
+    const min = this._minimum;
+    const max = this._maximum;
+
+    if (min !== null && max !== null) {
+      if (min > max) {
+        return;
+      }
+
+      const candidateStepValue = this._stepAlign(min + allowedValueStep, false);
+      if (candidateStepValue < min || candidateStepValue > max) {
+        return;
+      }
+    }
+
+    let value = this.valueAsNumber || 0;
```

Comment:
> Step 5. If applying the algorithm to convert a string to a number to the string given by the element's value does not result in an error, then let value be the result of that algorithm. Otherwise, let value be zero.

You will probably need to conver this line to a try catch.